wfe2/wfe.go

```diff
@@ -2331,6 +2337,12 @@ func (wfe *WebFrontEndImpl) NewOrder(
 		names[i] = ident.Value
 	}
 
+	err = policy.WellFormedDomainNames(names)
+	if err != nil {
+		wfe.sendError(response, logEvent, web.ProblemDetailsForError(err, "Invalid identifiers requested"), nil)
+		return
+	}
+
```

Comment:
Should we also add a check for `len(names) <= wfe.maxNames` here, so we can return that before getting into the rate limit transaction building?

Reply:
Good call. Note that moving this up changed the error message (tested in `test/integration/errors_test.go`) from `Error creating new order :: Order cannot contain more than 100 DNS names` to `Order cannot contain more than 100 DNS names` because we're generating the error within the WFE rather than generating it within RA (and then prepending a general error message). I think that's fine.

Reply:
Agreed. In `//ratelimits/bucket.go`, both `FailedAuthorizationsPerDomainPerAccountCheckOnlyTransactions` and `CertificatesPerDomainTransactions` document that they have a precondition of `len(orderDomains) < maxNames` and perform that bounds check internally, but the documentation leads me to believe it has been checked externally too.